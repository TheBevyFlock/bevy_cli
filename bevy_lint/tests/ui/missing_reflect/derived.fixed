//! Tests the `missing_reflect` lint when `Component`, `Resource`, and `Event` are derived.

#![feature(register_tool)]
#![register_tool(bevy)]
#![allow(dead_code)]
//~v NOTE: the lint level is defined here
#![deny(bevy::missing_reflect)]

use bevy::prelude::*;

//~v NOTE: `Component` implemented here
#[derive(Component)]
//~| HELP: `Reflect` can be automatically derived
//~v ERROR: defined a component without a `Reflect` implementation
#[derive(Reflect)]
struct MyComponent;

//~v NOTE: `Resource` implemented here
#[derive(Resource)]
//~| HELP: `Reflect` can be automatically derived
//~v ERROR: defined a resource without a `Reflect` implementation
#[derive(Reflect)]
struct MyResource {
    field_1: usize,
    field_2: bool,
}

//~v NOTE: `Event` implemented here
#[derive(Event)]
//~| HELP: `Reflect` can be automatically derived
//~v ERROR: defined an event without a `Reflect` implementation
#[derive(Reflect)]
struct MyEvent(String);
