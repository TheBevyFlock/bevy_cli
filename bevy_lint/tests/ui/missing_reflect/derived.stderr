error: defined an event without a `Reflect` implementation
  --> tests/ui/missing_reflect/derived.rs:30:1
   |
30 | struct MyEvent(String);
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
note: `Event` implemented here
  --> tests/ui/missing_reflect/derived.rs:27:10
   |
27 | #[derive(Event)]
   |          ^^^^^
note: the lint level is defined here
  --> tests/ui/missing_reflect/derived.rs:7:9
   |
7  | #![deny(bevy::missing_reflect)]
   |         ^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `Event` (in Nightly builds, run with -Z macro-backtrace for more info)
help: `Reflect` can be automatically derived
   |
30 + #[derive(Reflect)]
31 | struct MyEvent(String);
   |

error: defined a component without a `Reflect` implementation
  --> tests/ui/missing_reflect/derived.rs:15:1
   |
15 | struct MyComponent;
   | ^^^^^^^^^^^^^^^^^^^
   |
note: `Component` implemented here
  --> tests/ui/missing_reflect/derived.rs:12:10
   |
12 | #[derive(Component)]
   |          ^^^^^^^^^
   = note: this error originates in the derive macro `Component` (in Nightly builds, run with -Z macro-backtrace for more info)
help: `Reflect` can be automatically derived
   |
15 + #[derive(Reflect)]
16 | struct MyComponent;
   |

error: defined a resource without a `Reflect` implementation
  --> tests/ui/missing_reflect/derived.rs:21:1
   |
21 | / struct MyResource {
22 | |     field_1: usize,
23 | |     field_2: bool,
24 | | }
   | |_^
   |
note: `Resource` implemented here
  --> tests/ui/missing_reflect/derived.rs:18:10
   |
18 | #[derive(Resource)]
   |          ^^^^^^^^
   = note: this error originates in the derive macro `Resource` (in Nightly builds, run with -Z macro-backtrace for more info)
help: `Reflect` can be automatically derived
   |
21 + #[derive(Reflect)]
22 | struct MyResource {
   |

error: aborting due to 3 previous errors

